modules/csg/csg_shape.cpp

```diff
@@ -127,15 +127,12 @@ float CSGShape3D::get_snap() const {
 	return snap;
 }
 
-void CSGShape3D::_make_dirty() {
-	if (!is_inside_tree()) {
-		return;
+void CSGShape3D::_make_dirty(bool parent_removing) {
+	if ((parent_removing || is_root_shape()) && !dirty) {
+		call_deferred("_update_shape"); // Must be deferred; otherwise, is_root_shape() will use the previous parent
```

Comment:
Parameters start with `p_` in the C++ codebase.

```suggestion
void CSGShape3D::_make_dirty(bool p_parent_removing) {
	if ((p_parent_removing || is_root_shape()) && !dirty) {
		call_deferred("_update_shape"); // Must be deferred; otherwise, is_root_shape() will use the previous parent.
```